### DISCUSSION TITLE
Using "Offline" as a field value breaks my application??

### DISCUSSION THREAD

**Discussion by TheAyes** [supabase/supabase#30821] [Questions]
It took ages to debug this and I have no idea how to further proceed now.

I've come to the realization that the problem seems to be alot more specific. It's not that the "Type information" (in terms of intellisense) is wrong. But It seems there's someone else extremely broken.
My goal is to store the online status of users. Thus I made an enum type with offline, online and a few others. I defaulted to offline.
Which game me this error:
```
QWIK ERROR Code(3): Only primitive and object literals can be serialized [Function: LuX] Error: Code(3): Only primitive and object literals can be serialized
```
Which sure... is a qwik error but qwik isn't at fault here. For some reason online_status is returning a function. which is supposed to be used as a string?!

This function isn't serializable and thus crashes my application. What's interessting now is that I changed the datatype to a regular text. In Order to see if that fixes anything... which it didn't. What instead fixed it was changing the value inside of my rows from offline to something else (in my case test text). This resolved the serialization error but is rather undesired as I can't use "test text" as an actual status... I then decided to change it back to see if it was resolved, but the error came back upon changing test text back to offline.

Right now I'm unable to further proceed with development until I somehow resolve this issue...

My Table Definiton:
```sql
create table
  public.profiles (
    user_id uuid not null default auth.uid (),
    display_name text null default ''::text,
    updated_at timestamp with time zone not null default (now() at time zone 'utc'::text),
    avatar_url text null,
    account_name text not null default 'user'::text,
    bio text not null default ''::text,
    custom_status text null,
    online_status text not null default 'offline'::text,
    constraint profiles_pkey primary key (user_id),
    constraint profiles_account_name_key unique (account_name),
    constraint profiles_id_fkey foreign key (user_id) references auth.users (id) on delete cascade,
    constraint profiles_account_name_check check ((length(account_name) >= 3)),
    constraint profiles_display_name_check check ((length(display_name) > 1))
  ) tablespace pg_default;
  ```

**Comment by GaryAustin1** [comment]
A text field can only return text from Supabase.  The word offline is not anything special to Supabase or Postgres.  
I don't know Qwik and if it is using the REST API or SQL database port.  But even the PostgREST REST API supabase uses would still return a text value for a text field.
